### PR TITLE
Complete revamp of the module for 2023

### LINKED ISSUE #14: Create conditional for anycast IP setup
check demo project for how this was implemented, but basically 0/1 response (and thus the anycast_ip count) for setup and pass this option through to user-data like we did for secrets encryption.

this is functional in jmarhee/bgp-session-setup branch, but is not optional, so I won't be merging that as-is.

### PR DESCRIPTION
**Pull request by e-minguez**
Complete revamp of the module for 2023

Closes #14 #42

Fixes:
* Use templatefile instead the deprecated template_file
* Use metros instead the deprecated facilities
* Use the equinix terraform module instead the  deprecated equinix/metal
* Bird configuration with multiple peers
* sysctl changes persisted

Improvements:
* Multiple clusters on different or same metro with different topologies allowed
* No need for private ssh key, everything is done at cloud-init level
* K3s control plane HA via BGP
* Customization of different parameters (K3s version, MetalLB version, etc.)
* Demo app easily deployable
* Following the equinix/template standards
* Improved documentation
* More outputs to be easily consumable
* Customizable number of public IPv4 pool
* MetalLB deployment updated to use CRDs instead configmaps

### CODE REVIEW

File: example-ha-ip-pool-global-ip-example-multi-metro.tfvars [hunk 1]
```diff
@@ -0,0 +1,8 @@
+metal_auth_token = "redacted"
```
Comment (ctreatma) [inline example-ha-ip-pool-global-ip-example-multi-metro.tfvars:1]: The examples should each get their own directory in the `example` directory (e.g., this one could be in a directory named `example/ha-op-pool-global-ip-multi-metro/`); the example should include a `main.tf` that consumes the module. The directory structure for examples is demonstrated here: https://github.com/equinix-labs/terraform-equinix-template/tree/main/examples/simple

File: .terraform.lock.hcl [hunk 2]
```diff
@@ -1,132 +1,42 @@
 # This file is maintained automatically by "terraform init".
 # Manual edits may be lost in future updates.
 
-provider "registry.terraform.io/equinix/metal" {
-  version     = "2.1.0"
-  constraints = "~> 2.1"
+provider "registry.terraform.io/equinix/equinix" {
```
Comment (ctreatma) [inline .terraform.lock.hcl:4]: Since the terraform lock file is ignored by git, please remove the file from the repo rather than updating it.

**Review by ctreatma — COMMENTED** [review]
The CI is failing due to some formatting issues.  The `pre-commit` tool should be able to resolve those for you.  Could you try the following:

- [Install pre-commit](https://pre-commit.com/#install)
- Run `pre-commit install` to add pre-commit to git hooks
- Run `pre-commit run --all-files` to run all of the pre-commit checks (I'm not sure if this will commit the changes for you or if you will have to commit them manually)

File: example-ha-ip-pool-global-ip-example-multi-metro.tfvars (hunk 1 above)
Comment (e-minguez) [reply example-ha-ip-pool-global-ip-example-multi-metro.tfvars:1]: Added to the docs better

**Comment by e-minguez** [comment]
@ctreatma thanks for your inputs. I've sorted everything out. PTAL
/cc @dcallao

**Comment by dcallao** [comment]
@e-minguez Thanks for submitting these changes. I have tested this last week and will test again today one more time. Will update @displague for final approval then merge. 👍

**Comment by e-minguez** [comment]
> @e-minguez Thanks for submitting these changes. I have tested this last week and will test again today one more time. Will update @displague for final approval then merge. 👍

Do you want me to update my PR with @displague suggestions? or maybe we can merge this one and @displague can create a follow up PR on this repo? Open to suggestions 😄

**Comment by dcallao** [comment]
> > @e-minguez Thanks for submitting these changes. I have tested this last week and will test again today one more time. Will update @displague for final approval then merge. 👍
> 
> Do you want me to update my PR with @displague suggestions? or maybe we can merge this one and @displague can create a follow up PR on this repo? Open to suggestions 😄

Yes. Please do and thank you again. 👍

**Pull request merged**